src/handwritten/system/_collections/device_collection.py

```diff
@@ -59,6 +59,12 @@ def __getitem__(self, index):
             return [_DeviceAlternateConstructor(name, self._interpreter) for name in self.device_names[index]]
         elif isinstance(index, str):
             device_names = unflatten_channel_string(index)
+            all_devices = self.device_names
+            # Validate the device names we were provided
+            for device in device_names:
+                if device not in all_devices:
+                    raise KeyError(f'"{device}" is not a valid device name.')
```

Comment:
Terminology: `{device}` may be a perfectly valid device name. Most likely, it is a valid name but the device it refers to is not present.

In the DAQmx API DLLs, I think we would use this error: `kErrorDevCannotBeAccessed`

> Device cannot be accessed.  Possible causes:<LF/><LF/>Device is no longer present in the system.<LF/>Device is not powered.<LF/>Device is powered, but was temporarily without power.<LF/>Device and/or chassis driver support may have been removed.<LF/>Device is damaged.<LF/><LF/>Ensure the device is properly connected and turned on. Ensure the device and/or chassis is supported in the current version of the driver. Check the device's status in NI MAX. Turn the computer off and on again. If you suspect that the device is damaged, contact National Instruments at ni.com/support.

I don't think you should duplicate the full error message, but consider reusing part of it. "{device} cannot be accessed. Ensure the device is property connected and turned on."?